### PR TITLE
[SYNC-154][FE] Refresh token every time.

### PR DESCRIPTION
**Pull request by newb1er**
## Description: 
- call "user/refreshToken" action every time before sending token
- add "expirationTime" in needed
## Changes:
- add "user/refreshToken" action. call firebase api getIdToken() and update "user/token"
- reduce expiration time to 30 minutes
## Build Test:
✅

### CODE REVIEW

**Comment by newb1er** [comment]
now is able to refresh on profile without error message. 😭

File: src/store/user.js [hunk 1]
```diff
@@ -55,18 +58,25 @@ const actions = {
...
+          setExpiredTime(Date.now() + 30 * 60 * 1000)
           resolve(data.message)
         })
-        .catch(error => {
+        .catch((error) => {
           reject(error)
         })
     })
   },
+  async refreshToken({ commit }) {
```
Comment (jinan-tw) [inline src/store/user.js:73]: the function doesn't return a promise, but is invoked in `src/store/articles.js`, which may not know the error happened

**Review by jinan-tw — COMMENTED** [review]
`src/views/Profile.vue` 的解法不是很好，此修改在編輯文章編輯很久的時候送出，也有可能導致token過期的問題，希望能夠vuex的token隨時都是最新的狀態，並在API內當作參數傳到後端，可能再麻煩幫忙看看官方文件或線上教學，大家都怎麼處理token這段，送到後端的 Firebase Admin SDK

File: src/store/user.js [hunk 2]
```diff
@@ -55,18 +58,25 @@ const actions = {
   sendToken({ commit, dispatch }, userdata) {
     return new Promise((resolve, reject) => {
       login(userdata)
-        .then(response => {
+        .then((response) => {
           const { data } = response
           commit('SET_TOKEN', userdata.idToken)
           setToken(userdata.idToken)
-          setExpiredTime(Date.now() + 3 * 24 * 60 * 60 * 1000)
+          setExpiredTime(Date.now() + 30 * 60 * 1000)
```
Comment (jinan-tw) [inline src/store/user.js:65]: 這個目前已經沒有再使用了，Firebase自己有儲存的方式

File: src/store/user.js (hunk 1 above)
Comment (newb1er) [reply src/store/user.js]: removed. call 'getToken' instead.

File: src/store/user.js (hunk 2 above)
Comment (newb1er) [reply src/store/user.js]: removed.

**Comment by newb1er** [comment]
- 目前在Vuex放一個'getToken' action，需要token時會呼叫。
- 移除'token' state

**Pull request merged**